orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/model/AbstractStage.groovy

```diff
@@ -98,15 +93,7 @@ abstract class AbstractStage<T extends Execution> implements Stage<T>, Serializa
   }
 
   AbstractStage(Execution execution, String type) {
-    this(execution, type, EMPTY_MAP)
-  }
-
-  @Override
-  Stage preceding(String type) {
```

Comment:
This `preceding` is a horrible method ... only works consistently if the pipeline isn't parallel (which all should).